### DISCUSSION TITLE
websocket feature parity

### DISCUSSION THREAD

**Discussion by sfescape** [apache/pulsar#18756] [Q&A]
Is there a roadmap that discusses websocket features? It looks like websockets is pretty far behind in feature parity.
I'm considering moving from kafka to pulsar but topics regex consumer is an important feature for me.

**Comment by hpvd** [comment]
feature parity would indeed be pretty awesome and would imho help adoption further

**Comment by hpvd** [comment]
for latest state see: https://github.com/apache/pulsar/wiki/Client-Features-Matrix

**Comment by hpvd** [comment]
imho websocket is for pulsar something like "the window to the web world". It enables usage of pulsar for fields like web-tracking, live chat and many more...
If one see the potential of this community, one should give websocket bugs and feature-requests a fair priority in the next releases
While writing this there are roughly 25 issues open regarding websocket topics.
https://github.com/apache/pulsar/issues?q=is%3Aissue+is%3Aopen+websocket+
3 includes open pull request which may be easy to close.

Of course recourses are limited and not all fields have the same importance for adoption of pulsar.
But at least the points
- docs e.g. https://github.com/apache/pulsar/issues/8610, https://github.com/apache/pulsar/issues/8688
- security e.g. https://github.com/apache/pulsar/issues/3493
- performance e.g. https://github.com/apache/pulsar/issues/8584, https://github.com/apache/pulsar/issues/5979

are important for everyone thinking about using pulsar with websocket..
